tests/pull.bats

```diff
@@ -266,3 +266,49 @@ load helpers
   iid=$output
   run_buildah rmi ${iid}
 }
+
+@test "pull-policy" {
+  mkdir ${TESTDIR}/buildahtest
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --policy bogus alpine
+  expect_output --substring "unrecognized pull policy bogus"
+
+  #  If image does not exist the never will fail
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json --policy never alpine
+  expect_output --substring "no such image"
+  run_buildah 125 inspect alpine
+  expect_output --substring "image not known"
+
+  # create bogus alpine image
+  run_buildah from --signature-policy ${TESTSDIR}/policy.json scratch
+  cid=$output
+  run_buildah commit -q $cid docker.io/library/alpine
+  iid=$output
+
+  #  If image does not exist the never will succeed, but iid should not change
+  run_buildah pull --signature-policy ${TESTSDIR}/policy.json --policy never alpine
+  expect_output $iid
+
+  # Pull image by default should change the image id
+  run_buildah pull --policy always --signature-policy ${TESTSDIR}/policy.json alpine
+  if [[ $output == $iid ]]; then
+      expect_output "[output should not be '$iid']"
+  fi
```

Comment:
It would be good to have a second pull always here with a check that the pull happened rather than reusing the old local image.

Reply:
If the pull does not happen, then $output  would == the not alpine image and cause a failure.  That is the whole idea of the test.